scripts/docker-install.bash

```diff
@@ -24,11 +28,53 @@ git
 # just in case we want root
 sudo
 
+# For formatters installable from opam
+ocaml
+opam
+
+# For formatters installable from npm
+nodejs
+npm
+
+# For formatters installable from pip
+python3
+python3-pip
+
+# Formatters installable from apt
+# latexindent
+texlive-extra-utils
+# gofmt
+golang-go
+terraform
+"
+
+opam_packages="
+ocamlformat
+"
+
+npm_packages="
+prettier
+"
+
+pip_packages="
+black
+"
+
+stack_packages="
+brittany
 "
 
 export DEBIAN_FRONTEND=noninteractive
 apt-get update
-apt-get install -y $(grep -v "^#" <<< "$packages")
+apt-get install -y $(grep -v "^#" <<< "$apt_packages")
+opam install $(grep -v "^#" <<< "$opam_packages")
+npm install --global $(grep -v "^#" <<< "$npm_packages")
+pip install $(grep -v "^#" <<< "$pip_packages")
+# Stack
+curl -sSL https://get.haskellstack.org/ | sh
+stack install $(grep -v "^#" <<< "$stack_packages")
```

Comment:
You may find that you are unsatisfied with Stack-installed packages. They must be built from source and often take 15+ minutes to install, which is suboptimal for CI. Many Haskell projects provide binary downloads, which are often a better option.

If, despite our best efforts, we can't manage to make the Docker installation take a reasonable amount of time, I can suggest ways that it could be optimized. (My work on https://github.com/raxod502/riju gives me some context on dealing with installing poorly packaged software into Docker images.)

Reply:
Fixed with a nasty curl one-liner.